### PR TITLE
DOCS-2907-javamodule-duke

### PR DESCRIPTION
**Pull request by dukesphere**

### CODE REVIEW

**Review by alegmarra — APPROVED** [review]

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 1]
```diff
@@ -36,10 +44,24 @@ This example invokes a method that takes arguments:
...
+If the class is not compiled with this option, the same parameters must be
+referenced in the canonical way, as `arg0`, `arg1`, and so on. For the constructor
+to work with parameters that you reference by name, you also need to add the XML
+for the Maven compiler plugin to your `pom.xml` file for your Mule project.
+Otherwise, an error occurs.
+
+//EXAMPLE?
+Note that in Studio, the Java module provides DataSense to provide metadata
+associated with `<java:new/>` and both the output value and the input arguments.
+You can access the metadata
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc]: Where is the rest of this sentence?

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 2]
```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
 </java:invoke-static>
 ----
 
+For an example in Studio, see <<studio_example>>.
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc:78]: This lead-in seems unnecessary.

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 3]
```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
...
+
+[[studio_example]]
+=== Using New and Invoke in Studio
+
+In Studio, you can write or load a Java package, then configure Java operations
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc:83]: Remove  "then, and "finally"

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 4]
```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
...
+* Write two simple Java classes that contain instance methods: `Hello.java` and
+`Add.java`
+* Use the Maven compiler plugin to compile the classes in a way that allows for
+the use of named parameters in your configurations, instead of `arg0`, `arg1`,
+and so on.
+* Use the New operation in the Java module to instantiate the `Hello` and `Add`
+objects and to access named parameters in the instance methods.
+* Use the Invoke operation in the Java module to invoke the `hello()` and
+`add(3,4)` instance methods.
+* Run the Mule app to execute New and Invoke operations in Studio.
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc]: "to execute the New and Invoke operations"

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 5]
```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
...
+  }
+}
+----
++
+. Click the *javaexamples* tab in Studio to return to your Mule app, and set
+up a flow for `Hello` class and `hello()` method:
+.. In *javaexamples*, provide a trigger for a flow by dragging a Scheduler
+component into the Studio canvas.
++
+Optional: You can also set the Scheduler frequency to something less frequent
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc:225]: Maybe mention what the default is.

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 6]
```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
...
++
+Do not use the *_fx_* setting for the *Method* setting.
++
+.. Continuing with the New operation, click the *Advanced* configuration link, and
+set the *Output* to a target variable that stores the payload of the Invoke operation:
++
+* *Target Variable*: `mySum`
+* *Target Value*: `payload`
++
+This step simply shows how to pass the payload to a target variable if you
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc:281]: Remove "simply." Suggestion for the sentence:

This optional step shows how to pass the payload to a target variable.

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 7]
```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
...
+ever need to do so.
++
+.. Place a Logger component to the right of the Invoke operation in the
+new flow, click *fx*, and set the *Message* field to `vars.mySum`.
++
+This setting prints the payload stored in the target variable to the console.
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc:287]: I am not sure "print" is the best word here. Maybe "displays the payload...on the console."

File: modules/ROOT/pages/java/java-invoke-method.adoc [hunk 8]
```diff
@@ -54,9 +76,275 @@ You can invoke static Java methods like this:
...
++
+. Add XML for the Mule compiler plugin to the `pom.xml` file for your
+Mule project to make any named parameters readable.
++
+In the *Package Explorer* on the right, simply double-click `pom.xml`,
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-invoke-method.adoc]: Remove "simply"

File: modules/ROOT/pages/java/java-module.adoc [hunk 9]
```diff
@@ -4,24 +4,30 @@ include::_attributes.adoc[]
...
 * Avoid the need for you to know or understand Java.
 
-However, there are some advanced uses cases that require integrating with custom Java code, such as:
+However, there are some advanced uses cases that require integration with
+custom Java code, for example:
 
-* Need to reuse a tax calculation library.
-* Need to reuse a canonical object model that is standard in the organization.
-* Need to execute custom logic and Java is your weapon of choice.
+* Reuse of a library, such as tax calculation library.
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-module.adoc:15]: "the tax calculation library"

File: modules/ROOT/pages/java/java-module.adoc [hunk 10]
```diff
@@ -4,24 +4,30 @@ include::_attributes.adoc[]
...
+* Reuse of a canonical object model that is standard in the organization.
+* Execution of custom logic using Java.
 
-Note that in Mule 3, MEL was used as a bridge for creating instances of particular Java classes and executing Java methods. In Mule 4, interoperation with Java changes due to the replacement of MEL by DataWeave, a functional language. Since functional languages should not have side effects on their input arguments, it does not make sense for DataWeave to execute random instance methods on random object instances.
+Note that in Mule 3, MEL serves as bridge for creating instances of particular
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-module.adoc]: Is there an extra space before "operation"?

File: modules/ROOT/pages/java/java-module.adoc [hunk 11]
```diff
@@ -4,24 +4,30 @@ include::_attributes.adoc[]
...
+Java classes and executing Java methods. In Mule 4, interoperation with Java
```
Comment (dhirsch1) [inline modules/ROOT/pages/java/java-module.adoc:20]: It looks like there might be an extra space before "interoperation"

File: modules/ROOT/pages/java/java-invoke-method.adoc (hunk 1 above)
Comment (dukesphere) [reply modules/ROOT/pages/java/java-invoke-method.adoc]: that was removed some time ago, before this review. Strange that you're seeing it.

File: modules/ROOT/pages/java/java-invoke-method.adoc (hunk 5 above)
Comment (dukesphere) [reply modules/ROOT/pages/java/java-invoke-method.adoc:225]: The Scheduler isn't that relevant to this exercise. It's just there so they can run the app. The Java module operations are the focus.

File: modules/ROOT/pages/java/java-module.adoc (hunk 9 above)
Comment (dukesphere) [reply modules/ROOT/pages/java/java-module.adoc:15]: "a"  tax collection library

**Pull request merged**